### ISSUE TITLE
can't use amplify_authenticator: 0.1.0-rc.3 without global builder

### ISSUE THREAD

**Issue by quirkeer** [aws-amplify/amplify-flutter#1295]
For global usage it works fine.

```
  // Wrap your MaterialApp in an Authenticator component.
  return Authenticator(
    child: MaterialApp(
      theme: ThemeData.light(),
      darkTheme: ThemeData.dark(),

      // `builder` must be specified with `Authenticator.builder()`.
      builder: Authenticator.builder(),

      // You can use any combination of `routes`, `home`,
      // `onGenerateRoute`, etc. as long as `builder` is
      // configured as shown above.
      home: const LoggedInScreen(),
    ),
  );
```

But if I want to use the `Authenticator` in some widget, it can't work. But when I was using the version 0.1.0-rc.1, the issue won't happen.

```
  return MaterialApp(
      home: state.maybeWhen(
        unauthenticated: () => Authenticator(
          child: Text('need to login'),
        ),
        orElse: () => Center(),
      ),
    );
```

**Comment by Jordan-Nelson** [comment]
Hi @quirkeer - Thanks for the feedback.

We have work in progress to support protecting just a subset of routes with the authenticator, opposed to protecting every route. You can see a simple example of this in PR #1281.

Would this meet your use case? If this will not meet your use case (or if you aren't sure) would you mind explaining a little bit more about the use case?

**Comment by quirkeer** [comment]
That make sense, still got some questions:

- Does it mean that I still need to wrap the Material App by the Authenticator widget? 
- Does it support to use with package like `auto_route` together? 
- Is the behavior from `rc.1` right?
- Since I've tried not to use Authenticator on the top level but inner a child widget with `rc.1` version package, it works fine, should I change the logic or just keep the previous code?

**Comment by quirkeer** [comment]
@Jordan-Nelson Thank you anyway for help me~

**Comment by quirkeer** [comment]
could I use branch from you `Jordan-Nelson:authenticator/custom-ui` or if the new release will come soon? haha

**Comment by Jordan-Nelson** [comment]
> Does it mean that I still need to wrap the Material App by the Authenticator widget?

With rc.3 and up, you will need to wrap you MaterialApp with the authenticator component.

> Does it support to use with package like auto_route together?

I'm not too familiar with auto route, but I believe it would. I can take a look at the API for auto route and get back to you.

> Is the behavior from rc.1 right?

I'm not sure if I understand this question. My answer to the next question may answer this. Let me know if it does not.

> Since I've tried not to use Authenticator on the top level but inner a child widget with rc.1 version package, it works fine, should I change the logic or just keep the previous code?

Prior to rc.2, Authenticator was supposed to be a child of your Material App. Rc.3 changed that in order to fix an issue with routing. I expect this to be the API moving forward, but it is still subject to change while it's in release preview. If you are using rc.2 or earlier, you should keep it as is. You will need to change it for rc.3 and later.

I don't have an exact timeline. I can update this issue when the next RC is available with the changes in that PR. You can also follow that PR if you would like. I would expect us to have a new RC available to test shortly after that PR merges.

**Comment by quirkeer** [comment]
Thank you for your patience 💖 @Jordan-Nelson

**Comment by Jordan-Nelson** [comment]
Hey @quirkeer - Protecting a subset of routes is available in `0.1.0.rc.4`. You can find the docs for it [here](https://docs.amplify.aws/ui/auth/authenticator/q/framework/flutter/#unauthenticated-routes).

I have not had a chance to look at auto route and test this with that package yet, but plan to look at using it with popular navigation packages soon. Let me know if you have any questions or issues while using it. I'll leave this issue open for now if you have any feedback to provide while using it.

**Comment by Jordan-Nelson** [comment]
Hi @quirkeer - I tested the authenticator package with a handful of navigation packages, including auto_route. I did not have any issue using it with those packages.

The docs for unprotected routes can be found here: https://ui.docs.amplify.aws/components/authenticator?platform=flutter#unauthenticated-routes

I am going to close out this issue now.
